src/sentry/static/sentry/app/views/eventsV2/table.tsx

```diff
@@ -1,108 +1,56 @@
-// import Panel from 'app/components/panels/panel';
-// import LoadingContainer from 'app/components/loading/loadingContainer';
-// import EmptyStateWarning from 'app/components/emptyStateWarning';
-// import Placeholder from 'app/components/placeholder';
-// import {Form, SelectField, TextField} from 'app/components/forms';
-import GridResizable from 'app/components/gridResizable';
 
 import {DEFAULT_EVENT_VIEW_V1} from './data';
-// import {MetaType, getFieldRenderer} from './utils';
-import {MetaType} from './utils';
 import EventView from './eventView';
 import TableView from './tableView';
-import tableModalEditColumn from './tableModalEditColumn';
+import {TableData} from './tableTypes';
 
-type DataRow = {
-  [key: string]: string;
-};
-
-// TODO: move this
-type DataPayload = {
-  data: Array<DataRow>;
-  meta: MetaType;
-};
-
-type Props = {
+type TableProps = {
   api: Client;
   location: Location;
   organization: Organization;
 };
+type TableState = {
+  isLoading: boolean;
+  error: null | string;
 
-type State = {
   eventView: EventView;
-  loading: boolean;
-  error: null | string;
   pageLinks: null | string;
-  dataPayload: DataPayload | null | undefined;
+
+  tableData: TableData | null | undefined;
 };
 
 /**
- * Container element that fetches events and handles pagination
+ * `Table` is a container element that handles 2 things
+ * 1. Fetch data from source
+ * 2. Handle pagination of data
  */
-class Table extends React.PureComponent<Props, State> {
-  state: State = {
-    eventView: EventView.fromLocation(this.props.location),
-    loading: true,
-    error: null,
-    pageLinks: null,
-    dataPayload: null,
-  };
-
-  static getDerivedStateFromProps(props: Props, state: State): State {
+class Table extends React.PureComponent<TableProps, TableState> {
+  static getDerivedStateFromProps(props: TableProps, state: TableState): TableState {
     return {
       ...state,
       eventView: EventView.fromLocation(props.location),
     };
   }
 
-  static getColumnsFromQueryString(query: Location['query']) {
-    const {alias, field} = query;
-    const columnStrings: {
-      aggregationField: string;
-      name: string;
-    }[] = [];
-
-    if (typeof alias !== typeof field) {
-      throw new Error('Table.getColumnsFromQueryString: typeof alias !== typeof field');
-    } else if (typeof alias === 'string' && typeof field === 'string') {
-      columnStrings.push({aggregationField: field, name: alias});
-    } else if (
-      typeof alias === 'object' &&
-      typeof field === 'object' &&
-      alias !== null &&
-      field !== null
-    ) {
-      field.forEach((f, i) => {
-        columnStrings.push({aggregationField: f, name: alias[i]});
-      });
-    }
-
-    // columnStrings.map(column => {
+  state: TableState = {
+    isLoading: true,
+    error: null,
 
-    //   return {
-    //     name: '',
-    //     aggregation: '',
-    //     field: '',
+    eventView: EventView.fromLocation(this.props.location),
```

Comment:
Could the eventView be passed as props from the containing component in index.tsx?